api/src/main/java/org/pmiops/workbench/cohortreview/CohortReviewServiceImpl.java

```diff
@@ -85,8 +97,18 @@ public CohortReview saveCohortReview(CohortReview cohortReview) {
     }
 
     @Override
+    @Transactional
     public Iterable<ParticipantCohortStatus> saveParticipantCohortStatuses(List<ParticipantCohortStatus> participantCohortStatuses) {
-        return participantCohortStatusDao.save(participantCohortStatuses);
+        int i = 0;
+        for (ParticipantCohortStatus participantCohortStatus : participantCohortStatuses) {
+            entityManager.persist(participantCohortStatus);
+            i++;
+            if (i % batchSize == 0) {
+                entityManager.flush();
```

Comment:
You might want to add a log statement here.

Reply:
Done.